iSeg/ImageInformationDialogs.cpp

```diff
@@ -384,91 +387,61 @@ RotationDialog::RotationDialog(SlicesHandler* hand3D, QWidget* parent,
-	QObject::connect(le_dcc, SIGNAL(textChanged(const QString&)), this,
-					 SLOT(dcc_changed()));
-	QObject::connect(le_dcx, SIGNAL(textChanged(const QString&)), this,
-					 SLOT(dcx_changed()));
-	QObject::connect(le_dcy, SIGNAL(textChanged(const QString&)), this,
-					 SLOT(dcy_changed()));
-	QObject::connect(le_dcz, SIGNAL(textChanged(const QString&)), this,
-					 SLOT(dcz_changed()));
 
-	return;
+	QObject::connect(le_r11, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r12, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r13, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r21, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r22, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r23, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r31, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r32, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
+	QObject::connect(le_r33, SIGNAL(textChanged(const QString&)), this,
+					 SLOT(rotation_changed()));
 }
 
-RotationDialog::~RotationDialog() { delete vbox1; }
+RotationDialog::~RotationDialog() {}
 
-void RotationDialog::ParseValue(QLineEdit* le, unsigned short dcIndex)
+void RotationDialog::rotation_changed()
 {
-	bool b1;
-	float val = le->text().toFloat(&b1);
-	if (b1)
-	{
-		directionCosines[dcIndex] = val;
-	}
-	else
-	{
-		if (le->text() != QString(".") && le->text() != QString("-"))
-			QApplication::beep();
-	}
-}
-
-void RotationDialog::dca_changed()
-{
-	ParseValue(le_dca, 0);
-	pb_set->setEnabled(false);
-}
-
-void RotationDialog::dcb_changed()
-{
-	ParseValue(le_dcb, 1);
-	pb_set->setEnabled(false);
-}
-
-void RotationDialog::dcc_changed()
-{
-	ParseValue(le_dcc, 2);
-	pb_set->setEnabled(false);
-}
-
-void RotationDialog::dcx_changed()
-{
-	ParseValue(le_dcx, 3);
-	pb_set->setEnabled(false);
-}
-
-void RotationDialog::dcy_changed()
-{
-	ParseValue(le_dcy, 4);
-	pb_set->setEnabled(false);
-}
-
-void RotationDialog::dcz_changed()
-{
-	ParseValue(le_dcz, 5);
-	pb_set->setEnabled(false);
+	pb_set->setEnabled(true);
 }
 
 void RotationDialog::set_pressed()
 {
-	bool ok[6];
-	le_dca->text().toFloat(&ok[0]);
-	le_dcb->text().toFloat(&ok[1]);
-	le_dcc->text().toFloat(&ok[2]);
-	le_dcx->text().toFloat(&ok[3]);
-	le_dcy->text().toFloat(&ok[4]);
-	le_dcz->text().toFloat(&ok[5]);
-	bool tmp = true;
-	for (unsigned short i = 0; i < 6; ++i)
+	QLineEdit* les[] = {
+		le_r11, le_r12, le_r13,
+		le_r21, le_r22, le_r23,
+		le_r31, le_r32, le_r33
+	};
+
+	bool ok = true;
+	for (int i=0; i<9; i++)
 	{
-		tmp &= ok[i];
+		if (ok) les[i]->text().toFloat(&ok);
 	}
-	if (tmp)
+	if (ok)
 	{
+		rotation[0][0] = le_r11->text().toFloat();
```

Comment:
just wondering, why do you use a QTextEdit and not a SpinbBox or a QTextEdit with a QValidator? this would prevent all this checking code for correct floating point value (e.g. QDoubleValidator)

Reply:
good point, thanks